src/alignment.rs

```diff
@@ -199,50 +199,50 @@ impl Alignment {
                 _ => {
                     x_i = self.xstart;
                     y_i = self.ystart;
-                    for k in x.iter().take(self.xstart) {
-                        x_pretty.push_str(&format!("{}", String::from_utf8_lossy(&[*k])));
-                        inb_pretty.push(' ');
-                        y_pretty.push(' ')
+                    // SAFETY: validity checked before creating an Alignment
+                    unsafe {
+                        x_pretty.push_str(std::str::from_utf8_unchecked(&x[..self.xstart]));
                     }
-                    for k in y.iter().take(self.ystart) {
-                        y_pretty.push_str(&format!("{}", String::from_utf8_lossy(&[*k])));
-                        inb_pretty.push(' ');
-                        x_pretty.push(' ')
+                    y_pretty.push_str(&" ".repeat(self.xstart));
+                    unsafe {
+                        y_pretty.push_str(std::str::from_utf8_unchecked(&y[..self.ystart]));
                     }
+                    x_pretty.push_str(&" ".repeat(self.ystart));
+                    inb_pretty.push_str(&" ".repeat(self.xstart + self.ystart));
```

Comment:
I'm not too fond of unnecessary `unsafe`s, and I doubt performance is crucial for pretty printing alignments (disregarding the fact that this might not even be faster).
However, since `TextSlice` is just an alias for `&'a [u8]` and  `u8 as char` [though confusing, usually works as expected](https://doc.rust-lang.org/src/core/char/convert.rs.html#115-150), `x_pretty.push_str(...)` may be replaced by `&x[..self.xstart].for_each(|x| x_pretty.push(x as char))` or so.